### DISCUSSION TITLE
Is it possible to recalculate DVH parameters for an image, using the exact plan, precalculated and optimized for another image?

### DISCUSSION THREAD

**Discussion by mlapaeva** [e0404/matRad#638] [Q&A]
Dear MatRad team,

I would like to kindly ask, whether it is possible via MatLab functions to calculate the DVH parameters for an image, based on pre-calculated and pre-optimised plan on another?
We are working on synthetic CT projecs, so, we would like to run calculations and optimisation of the plan with the fixed set of parameters on original image and then, just update the actual image array with synthetic CT image, while applying the exactly the same plan parameters. (and we would like to do it for many patients, so, looking for automation with your great tool).

I have figured out how to have a fixed set of parameters for all plans with:
1) selecting radiation mode https://github.com/e0404/matRad/wiki/How-to-execute-matRad#radMode
2) definition of objective and constrains: https://github.com/e0404/matRad/discussions/517

and now I am wondering what I need to save to apply the same plan, which was previously calculated and optimised on original CT, while I just want to apply it to sCT and further calculate the DVH discrepancies in between of original and synthetic.

Regards,
Mariya

**Comment by remocristoforetti** [comment]
Dear Mariya,
if I understand correctly your setup and what you want to do I would suggest following these steps:

- First load the original CT image into **matRad**, setup the parameters of the plan you would like to optimize. For this you can have a look at the examples or ask here any further suggestion.

- Run the `dij `calculation and the fluence optimization for this patient setup. As an output of `matRad_fluenceOptimization ` you will find both the physicalDose/RBE/LET distributions (according to the parameters of the plan you setup). In addition to that, you will also find the list of optimized fluence weights (usually as `resultGUI.w` if running **matRad** form script).

- You can then load the new sCT into **matRad**. If the image sets are compatible (same number of CT slices, same Isocenter position and so on), it should be sufficient to perform the dose calculation (i.e obtain the `dij ` variable) for the new sCT dataset (via `matRad_calcParticleDose `or `matRad_calcPhotonDose`, according to the specific radiation modality).

- Once the dose calculation has been performed for the new sCT dataset, there is no need to re-optimize the fluence weight, you could simply run the function `matRad_calcDoseDirect` and provide the set of previously optimized weights as an input. That should allow to compare the same plan configuration applied to different CTs/patient setups.

Than the DVHs can be easily computed and visualized through **matRad** functionalities (like `matRad_calcDVH `and `matRad_showDVH `for example).

Please let me know if you need any further help on setting this up.

Best regards,
Remo

**Reply by mlapaeva** [reply]
Thank you a lot, Remo! 
It gets much more clear! I have managed to import all of my dicoms and save them to .mat format via matlab scripts with the required set of objectives and constrains, as well as compute the plan for original image with the beam\etc parameters we need. Before loading and computing the plan for sCTs I would like to kindly ask you:
How can I update the number of stratification levels?
![image](https://github.com/e0404/matRad/assets/34272754/2dbcbc18-a1d6-4750-85bb-e37bbac2e022)
I have tried to update 
`pln.numOfLevels=5; `
It has not helped so far.

Regards,
Mariya

**Reply by wahln** [reply]
Which version / branch of matRad are you using? I think in the Release 2.10.1 there is unfortunately still no pln-parameter to store the sequencing/straticifaction levels. Instead, it is manually passed to the sequencing function (check out the matRad.m) script:
https://github.com/e0404/matRad/blob/d6ea09a29307d38b231ef5d3b379c3b9af9a4f05/matRad.m#L69-L74

In the GUI this is handled behind the scenes, and in the future there will be a dedicated parameter for this also storable in the pln.
